Add 'Back-to-top' button for partners list page

lalit97 opened the pull request:
The partners page is quite long now. A 'back to top' button was needed.

Bug: T170113

Samwalton9 commented:
Could you add a screenshot of what this button looks like, please? :)

lalit97 commented:
![back-to-top2](https://user-images.githubusercontent.com/31045288/75144878-f12abf00-571d-11ea-998f-f237f4100bca.gif)

lalit97 commented:
Hello @Samwalton9 I thought you just missed this one, please have a look whenever you are free :)

Samwalton9 commented:
Hey lalit - thanks for the reminder. I'm actually going to wait on reviewing this. We've got a lot of moving pieces being merged in to the tool this month and it might just reduce complications to hold off on merging this PR until then. Sorry about that - we'll get to it in a few weeks :)

lalit97 commented:
Okay, no problem :)

jsnshrmn commented:
I rebased this so I can check it against our current code base and review.

jsnshrmn commented on TWLight/static/css/local.css:
```diff
@@ -748,3 +748,28 @@ CLEARING COLUMNS
+#back-to-top.display{
+  opacity: 1;
+  visibility: visible;
+}
```
files should end with a newline

lalit97 replied:
added a newline :)

jsnshrmn commented:
Also, we've moved to an anonymizing reverse proxy for external assets, so it would be great if you updated the URLs to match what is in #447. Since you had no way of knowing that this change was about to happen, I'd be happy to make that update on your behalf if you don't want to.

lalit97 commented:
> This occludes the footer navigation in mobile views. That needs to be resolved

Fixed it, now working well in mobile view.

> When I load the page without JavaScript, this feature disappears. Please use JavaScript for progressive enhancement where possible. Since this is just navigation between page anchors, it should work without JavaScript.

removed the javascript part.

lalit97 commented:
> Also, we've moved to an anonymizing reverse proxy for external assets, so it would be great if you updated the URLs to match what is in #447. Since you had no way of knowing that this change was about to happen, I'd be happy to make that update on your behalf if you don't want to.

I have removed the urls, since they were already present in `base.html`. Thank you for such a detailed review. I have fixed all the points you have mentioned :)

suecarmol commented on TWLight/resources/templates/resources/partner_filter.html:
```diff
@@ -49,4 +49,8 @@ <h2>{% trans "Browse Partners" %}</h2>
     {% endfor %}
   </div>
+
+  <a id="back-to-top" href="#" class="btn btn-light btn-sm">
```
Can you add a tooltip that says `Back to top` along with its corresponding translator comment?

lalit97 replied:
added a tooltip :)

suecarmol commented on TWLight/static/css/local.css:
```diff
@@ -752,3 +753,24 @@ CLEARING COLUMNS
+  display: inline-block;
+  border-radius: 4px;
+  position: fixed;
+  bottom: 210px;
```
I'm going back and forth on whether this should be lower (around 170px). At 170px, the button doesn't cover the footer in any device I have tested it on. Is there a reason why you chose 210px?

lalit97 replied:
Reduced the value to 170px you are right it is working fine on this value at all devices :)
